wwwapp/models.py

```diff
@@ -281,3 +302,30 @@ def result_in_percent(self):
 
     class Meta:
         unique_together = [('workshop', 'participant')]
+
+
+class ResourceYearPermission(models.Model):
+    """
+    Resource associated with a WWW edition (year). Resource can be accessed by
+    users who qualified or had a lecture on year. User is granted access to
+    root_url and recursively to all files and subdirectories inside.
+    """
+    display_name = models.CharField(max_length=50, blank=True)
+    access_url = models.URLField(blank=True,
+                                 help_text="URL dla przycisku w menu. Przycisk nie jest wyświetlany jeśli url jest pusty")
+    root_url = models.CharField(max_length=256, null=False, blank=False,
+                                help_text='URL bez protokołu, bez "/" na końcu. np. "internet.warsztatywww.pl/www15"')
```

Comment:
I would add validation of the format in clean(), especially the trailing / rule because that's easy to miss